src/main/java/org/aksw/gerbil/http/InterruptingObserver.java

```diff
@@ -58,14 +58,20 @@ private void checkAnnotators() {
                 waitingTime = currentTime - e.getValue();
                 if (waitingTime > maxWaitingTime) {
                     observedRequest = e.getKey();
-                    LOGGER.info("The HTTP request emitter \"{}\" already runs for {} ms. Trying to interrupt it.",
-                            observedRequest.emitter.getName(), waitingTime);
-                    try {
-                        observedRequest.emitter.interrupt(observedRequest.request);
-                    } catch (UnsupportedOperationException ex) {
-                        LOGGER.error("Couldn't interrupt request of HTTP request emitter \""
-                                + observedRequest.emitter.getName() + "\" that is already running for " + waitingTime
-                                + " ms.");
+                    if (observedRequest.request.isAborted()) {
+                        LOGGER.warn(
+                                "Observing an HTTP request that is already aborted. This could be mean that the HTTP management does not have been informed about the termination of a request.");
+                        observedRequests.remove(observedRequest);
```

Comment:
I think this should throw a `ConcurrentModificationException`. Maybe use the `iterator` of the `entrySet()` and then the
`remove` method of the iterator?

Reply:
Fixed it as suggested.